src/averageAges.js

```diff
@@ -52,7 +68,24 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const ageData = people.reduce((acc, person) => {
+    const criteria = onlyWithSon ? person.sex === 'm' : true;
+
+    if (criteria) {
+      const mother = people.filter((human) => human.name === person.mother)[0];
+
+      if (mother) {
+        acc.totalAge += person.born - mother.born;
+        acc.mothersCount += 1;
+      }
+    }
+
+    return acc;
+  }, {
+    mothersCount: 0, totalAge: 0,
+  });
```

Comment:
You have chosen a little bit long and complicated way to solve this part of the task.
A more simple variant (and without any 'if-statements' and counters) would be to create an array of children according to 'onlyWithSon' argument using method `.some` inside '.filter'.
Then `.map` this array to find the age difference between mother and child (finding the child's mother using `.find` method inside `.map`.
And then just calculate the average difference of their ages using our new array and its length.